src/main/scala/beam/agentsim/agents/planning/BeamPlan.scala

```diff
@@ -41,6 +45,36 @@ object BeamPlan {
     beamPlan.createToursFromMatsimPlan()
     beamPlan
   }
+
+  def addOrReplaceLegBetweenActivities(plan: Plan, leg: Leg, originActivity: Activity, destinationActivity: Activity): Plan = {
+    val newPlanElements = plan.getPlanElements.asScala.sliding(2).map{ elems =>
```

Comment:
This isn't dire to change, but pattern matching would resolve some of the casts and clean up the code here.